ecs-task-definition-deploy/actions_helper/main.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+from dataclasses import dataclass
+
+import boto3
+import click
+from botocore.client import BaseClient
+
+
+class NonSingleValueError(Exception):
+    pass
+
+
+@dataclass
+class Tag:
+    key: str
+    value: str
+
+
+def get_tags(task_definition_tags: str) -> tuple[Tag, ...]:
+    try:
+        return tuple(
+            Tag(tag.strip().split(":")[0], tag.strip().split(":")[1])
+            for tag in task_definition_tags.split(",")
+        )
+    except IndexError:
+        raise ValueError(
+            "Invalid task definition tag format, expects 'key:value,key:value'"
+        )
+
+
+def get_active_task_definition_arn_by_tag(
+    *,
+    task_definition_family_prefix: str,
+    task_definition_tags: str,
+    allow_empty: bool,
+) -> str:
+    ecs_client: BaseClient = boto3.Session(region_name="eu-central-1").client("ecs")
+
+    tags = get_tags(task_definition_tags)
+
+    active_task_definition_arns = ecs_client.list_task_definitions(
+        familyPrefix=task_definition_family_prefix,
+        status="ACTIVE",
+        sort="DESC",
+    )["taskDefinitionArns"]
+
+    active_task_definitions = [
+        task_definition["taskDefinition"]["taskDefinitionArn"]
+        for task_definition_arn in active_task_definition_arns
+        if (
+            task_definition := ecs_client.describe_task_definition(
+                taskDefinition=task_definition_arn,
+                include=["TAGS"],
+            )
+        )
+        and all(
+            {"key": tag.key, "value": tag.value} in task_definition["tags"]
+            for tag in tags
+        )
+    ]
+
+    if allow_empty and not active_task_definitions:
+        return ""
```

Comment:
Can you explain your reasoning for `allow_empty`? It seems you are setting it to true in the action. Naively, I would think that not finding exactly one active task definition with the tag should always be an exception.

Reply:
The first time a deployment is made
The script will fail because at that time there wont be a task definition with the tag `created_by: Github Actions Deployment`

So the script needs to configurably allow for empty result, by default it's False